Reword and clarify embedding on views (#560)
begriffs/postgrest-docs: docs/api.rst

Embedding Views
---------------

> Embedding a view is possible if its columns have **foreign keys** defined in their source table. Source tables are the ones referenced in the ``FROM`` clause and the multiple ``JOIN`` clauses.

Reviewer: ```suggestion
Embedding a view is possible if its columns have **foreign keys** defined in their source tables. Source tables are the ones referenced in the ``FROM`` clause and the multiple ``JOIN`` clauses.
```

> Embedding a view is possible if its columns have **foreign keys** defined in their source tables. Source tables are the ones referenced in the ``FROM`` clause and the multiple ``JOIN`` clauses.

Reviewer: ```suggestion
PostgREST will infer the relationships of a view based on its source tables. Source tables are the ones referenced in the FROM and JOIN clauses of the view definition. The foreign keys of the relationships must be present in the top SELECT clause of the view for this to work.
```

> For instance, the following view has ``nominations``, ``films``and ``competitions`` as source tables:

Reviewer: ```suggestion
For instance, the following view has ``nominations``, ``films`` and ``competitions`` as source tables:
```

.. code-block:: postgres

  CREATE VIEW nominations_view AS
    SELECT
       films.title as film_title
     , competitions.name as competition_name
     , nominations.rank
     , nominations.film_id as nominations_film_id
     , films.id as film_id
    FROM nominations
    JOIN films ON films.id = nominations.film_id
    JOIN competitions ON competitions.id = nominations.competition_id;

> Since this view contains ``nominations.film_id`` — with a **foreign key** defined in its source table — then we can embed the ``films`` table. Similarly, because the view contains ``films.id``, then we can also embed the ``roles`` and the ``actors`` tables (the last one in a many-to-many relationship):

Reviewer: ```suggestion
Since this view contains ``nominations.film_id``, which has a **foreign key** relationship to ``films``, then we can embed the ``films`` table. Similarly, because the view contains ``films.id``, then we can also embed the ``roles`` and the ``actors`` tables (the last one in a many-to-many relationship):
```

Embedding Views
---------------

> Embedding a view is possible if its columns have **foreign keys** defined in their source table, which, in turn, must be clearly present in the ``FROM`` clause of the view definition (without using sub queries).

Reviewer: I've been thinking of a further simplification. How about if we refer to source tables as the ones in the FROM/JOIN targets? So in `FROM (SELECT * FROM other_table)`, `other_table` wouldn't be a source table for us.

Reviewer: So it'd be:
```suggestion
Embedding a view is possible if its columns have **foreign keys** defined in their source table. Source tables are the tables referenced in the ``FROM`` and ``JOIN`` clauses.
```
